common/common.go

```diff
@@ -94,6 +94,42 @@ const (
 	TCPFlagCwr = 0x80
 )
 
+// ErrorCode type for codes of errors
+type ErrorCode int
+
+// constants with error codes
+const (
+	CodeFail ErrorCode = iota + 1
+	CodeParseCPUListFail
+	CodeReqTooManyPorts
+	CodeBadArgument
+	CodeUseNilFlow
+	CodeOpenedFlowAtTheEnd
+	CodePortHasNoQueues
+	CodeNotAllQueuesUsed
+	CodeFailToInitPort
+	CodeParseRuleJSONErr
+	CodeFileError
+	CodeParseRuleError
+	CodeIncorrectArgInRules
+	CodeIncorrectRule
+	CodeAllocMbufError
+	CodePktMbufHeadRoomTooSmall
+	CodeNotEnoughCores
+	CodeNotFound
+)
+
+// NFError is error type returned by yanff functions
+type NFError struct {
+	Code    ErrorCode
+	Message string
+}
+
+// Error method to implement error interface
+func (err NFError) Error() string {
+	return fmt.Sprintf("failed with message '%s' and code '%d'", err.Message, err.Code)
```

Comment:
The rule of thumb for error messages for `Error` is that they should be easy to use in any logging scheme by end-user.
Proposing:
```go
return fmt.Sprintf("%s (%d)", err.Message, err.Code)
```
Let the user choose what to quote, or what quotes to use.
For example, your error could be used like this:
```go
if err != nil {
	log.Fatalf("failed at %v: yanff error='%v'", someContext, err)
}
```
Note the single quotes that wrap the error. With quotes inside Error return, output will be more ugly.